<a href="https://colab.research.google.com/github/FelipeGonzalezGarrido/Coding-Dojo/blob/main/Ejercicio_de_altura_promedio.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [4]:
from google.colab import drive
drive.mount ("/content/drive")
import pandas as pd

Mounted at /content/drive


In [5]:
filename = "/content/drive/MyDrive/Data CodingDojo/athleteEventsNoPersonal.csv"

In [6]:
df = pd.read_csv(filename)

Analisis primario

info, shape , describe

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40616 entries, 0 to 40615
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      40616 non-null  int64  
 1   Age     39143 non-null  float64
 2   Height  31615 non-null  float64
 3   Team    40616 non-null  object 
 4   NOC     40616 non-null  object 
 5   Games   40616 non-null  object 
 6   Year    40616 non-null  int64  
 7   Season  40616 non-null  object 
 8   City    40616 non-null  object 
 9   Sport   40616 non-null  object 
 10  Event   40616 non-null  object 
 11  Medal   5917 non-null   object 
dtypes: float64(2), int64(2), object(8)
memory usage: 3.7+ MB


In [8]:
df.shape

(40616, 12)

In [9]:
df.describe(include = "all")

ID           Age        Height           Team    NOC  \
count    40616.000000  39143.000000  31615.000000          40616  40616   
unique            NaN           NaN           NaN            646    222   
top               NaN           NaN           NaN  United States    USA   
freq              NaN           NaN           NaN           2530   2668   
mean     68140.888738     25.619498    175.381117            NaN    NaN   
std      38830.405762      6.494926     10.561791            NaN    NaN   
min          5.000000     11.000000    135.000000            NaN    NaN   
25%      34661.750000     21.000000    168.000000            NaN    NaN   
50%      68016.500000     25.000000    175.000000            NaN    NaN   
75%     101475.000000     28.000000    183.000000            NaN    NaN   
max     135571.000000     74.000000    221.000000            NaN    NaN   

              Games          Year  Season    City      Sport  \
count         40616  40616.000000   40616   40616      40616   
unique           51           NaN       2      42         60   
top     2008 Summer           NaN  Summer  London  Athletics   
freq           2056           NaN   33081    3370       5550   
mean            NaN   1978.653240     NaN     NaN        NaN   
std             NaN     29.850461     NaN     NaN        NaN   
min             NaN   1896.000000     NaN     NaN        NaN   
25%             NaN   1960.000000     NaN     NaN        NaN   
50%             NaN   1988.000000     NaN     NaN        NaN   
75%             NaN   2004.000000     NaN     NaN        NaN   
max             NaN   2016.000000     NaN     NaN        NaN   

                          Event Medal  
count                     40616  5917  
unique                      726     3  
top     Football Men's Football  Gold  
freq                        845  2009  
mean                        NaN   NaN  
std                         NaN   NaN  
min                         NaN   NaN  
25%                         NaN   NaN  
50%                         NaN   NaN  
75%                         NaN   NaN  
max                         NaN   NaN

In [10]:
df.head()

ID   Age  Height         Team  NOC        Games  Year  Season         City  \
0   5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter      Calgary   
1   5  21.0   185.0  Netherlands  NED  1988 Winter  1988  Winter      Calgary   
2   5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter  Albertville   
3   5  25.0   185.0  Netherlands  NED  1992 Winter  1992  Winter  Albertville   
4   5  27.0   185.0  Netherlands  NED  1994 Winter  1994  Winter  Lillehammer   

           Sport                               Event Medal  
0  Speed Skating    Speed Skating Women's 500 metres   NaN  
1  Speed Skating  Speed Skating Women's 1,000 metres   NaN  
2  Speed Skating    Speed Skating Women's 500 metres   NaN  
3  Speed Skating  Speed Skating Women's 1,000 metres   NaN  
4  Speed Skating    Speed Skating Women's 500 metres   NaN

In [11]:
df.isna().sum()

ID            0
Age        1473
Height     9001
Team          0
NOC           0
Games         0
Year          0
Season        0
City          0
Sport         0
Event         0
Medal     34699
dtype: int64

# 1.1 Hay 40616 filas
# 1.2 hay 12 columnas
# 1.3 las columnas age, height y medal presentan nulos.
## la columna medal posee esa cantidad de nulos debido a  que por competencia se entregan 3 medallas, los demas competidores no reciben medalla.
# 1.4 Londres aparece en 3370  registros


In [12]:
df[df["City"] == "London"].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3370 entries, 36 to 40609
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      3370 non-null   int64  
 1   Age     3067 non-null   float64
 2   Height  2139 non-null   float64
 3   Team    3370 non-null   object 
 4   NOC     3370 non-null   object 
 5   Games   3370 non-null   object 
 6   Year    3370 non-null   int64  
 7   Season  3370 non-null   object 
 8   City    3370 non-null   object 
 9   Sport   3370 non-null   object 
 10  Event   3370 non-null   object 
 11  Medal   568 non-null    object 
dtypes: float64(2), int64(2), object(8)
memory usage: 342.3+ KB


# 1.5 El atleta mas joven tiene 11 anios


In [13]:
df["Age"].min()

11.0

# 2.1 Cuantos atletas que participaron en el deporte tug of war pertenecian en el equipo sweden en el conjunto

6 atletas

In [14]:
df[(df["Sport"] == "Tug-Of-War") & (df["Team"] == "Sweden")]

ID   Age  Height    Team  NOC        Games  Year  Season    City  \
10023  34205  22.0     NaN  Sweden  SWE  1908 Summer  1908  Summer  London   
12681  43105  23.0     NaN  Sweden  SWE  1906 Summer  1906  Summer  Athina   
13015  44087  29.0   182.0  Sweden  SWE  1906 Summer  1906  Summer  Athina   
16250  55169  23.0     NaN  Sweden  SWE  1908 Summer  1908  Summer  London   
19048  64266  21.0     NaN  Sweden  SWE  1908 Summer  1908  Summer  London   
20550  68647  26.0   190.0  Sweden  SWE  1906 Summer  1906  Summer  Athina   

            Sport                        Event   Medal  
10023  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN  
12681  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze  
13015  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze  
16250  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN  
19048  Tug-Of-War  Tug-Of-War Men's Tug-Of-War     NaN  
20550  Tug-Of-War  Tug-Of-War Men's Tug-Of-War  Bronze

# 2.2      b. ¿Cuántas medallas de oro se le entregaron a los atletas que tenían más de 40 años?

se entregaron 76 medallas de oro a atletas de 40 o mas anios

In [15]:
df[(df["Age"] >= 40)&(df["Medal"] == "Gold")]

ID   Age  Height                Team  NOC        Games  Year  \
740      2735  41.0   168.0              Russia  RUS  2000 Summer  2000   
1103     4029  46.0     NaN           Argentina  ARG  1936 Summer  1936   
1421     4993  41.0   180.0             Mosk II  NOR  1920 Summer  1920   
3040    10769  49.0   180.0          White Lady  DEN  1964 Summer  1964   
3270    11617  44.0     NaN             Heroine  GBR  1908 Summer  1908   
...       ...   ...     ...                 ...  ...          ...   ...   
38916  130271  43.0     NaN            Heira II  NOR  1920 Summer  1920   
38952  130381  42.0   167.0       United States  USA  1972 Summer  1972   
39193  130936  56.0     NaN       United States  USA  1908 Summer  1908   
39197  130936  60.0     NaN       United States  USA  1912 Summer  1912   
39365  131533  58.0     NaN  Cincinnati Archers  USA  1904 Summer  1904   

       Season       City             Sport  \
740    Summer     Sydney          Shooting   
1103   Summer     Berlin              Polo   
1421   Summer  Antwerpen           Sailing   
3040   Summer      Tokyo           Sailing   
3270   Summer     London           Sailing   
...       ...        ...               ...   
38916  Summer  Antwerpen           Sailing   
38952  Summer     Munich           Archery   
39193  Summer     London          Shooting   
39197  Summer  Stockholm  Art Competitions   
39365  Summer  St. Louis           Archery   

                                             Event Medal  
740    Shooting Men's Rapid-Fire Pistol, 25 metres  Gold  
1103                               Polo Men's Polo  Gold  
1421                       Sailing Mixed 10 metres  Gold  
3040           Sailing Mixed Three Person Keelboat  Gold  
3270                        Sailing Mixed 7 metres  Gold  
...                                            ...   ...  
38916                      Sailing Mixed 12 metres  Gold  
38952                   Archery Women's Individual  Gold  
39193   Shooting Men's Running Target, Double Shot  Gold  
39197           Art Competitions Mixed Sculpturing  Gold  
39365                   Archery Women's Team Round  Gold  

[76 rows x 12 columns]

# 3. Determinar la altura promedio de:

#3.1 todas las filas incluidas
La altura promedio de todas las filas incluidas es 1,75 metros

In [16]:
df["Height"].mean().round()

175.0

#3.2 Todas las filas incluidas, pero agrupadas por Event

In [17]:
df.groupby("Event")["Height"].mean()

Event
Alpine Skiing Men's Combined                      178.547170
Alpine Skiing Men's Downhill                      178.211679
Alpine Skiing Men's Giant Slalom                  176.739394
Alpine Skiing Men's Slalom                        176.482143
Alpine Skiing Men's Super G                       179.850575
                                                     ...    
Wrestling Women's Flyweight, Freestyle            158.100000
Wrestling Women's Heavyweight, Freestyle          173.444444
Wrestling Women's Light-Heavyweight, Freestyle    170.000000
Wrestling Women's Lightweight, Freestyle          162.666667
Wrestling Women's Middleweight, Freestyle         164.533333
Name: Height, Length: 726, dtype: float64

In [18]:
df["ID"].value_counts

<bound method IndexOpsMixin.value_counts of 0             5
1             5
2             5
3             5
4             5
          ...  
40611    135567
40612    135570
40613    135570
40614    135571
40615    135571
Name: ID, Length: 40616, dtype: int64>

# 3.3 Promedio de altura de atletas sin incluir ID repetidos

1,76 metros

In [21]:
#Conteo de duplicados
df.duplicated().sum()

207

In [20]:
# se aisla los ID unico en nueva variable, donde por columna y argumento first, se mantiene el primer ID y se desechan los repetidos
one_per_athlete = df.drop_duplicates(subset = 'ID', keep = 'first')
one_per_athlete

ID   Age  Height           Team  NOC        Games  Year  Season  \
0           5  21.0   185.0    Netherlands  NED  1988 Winter  1988  Winter   
6           6  31.0   188.0  United States  USA  1992 Winter  1992  Winter   
14         12  31.0   172.0        Finland  FIN  2000 Summer  2000  Summer   
15         30  26.0   189.0    Netherlands  NED  1996 Summer  1996  Summer   
17         36  21.0   194.0    Netherlands  NED  1996 Summer  1996  Summer   
...       ...   ...     ...            ...  ...          ...   ...     ...   
40604  135553  21.0   168.0   Soviet Union  URS  1952 Summer  1952  Summer   
40609  135558  32.0   200.0         Poland  POL  2012 Summer  2012  Summer   
40610  135567  24.0   183.0         Russia  RUS  2000 Summer  2000  Summer   
40612  135570  27.0   176.0         Poland  POL  2014 Winter  2014  Winter   
40614  135571  30.0   185.0         Poland  POL  1998 Winter  1998  Winter   

              City                 Sport  \
0          Calgary         Speed Skating   
6      Albertville  Cross Country Skiing   
14          Sydney             Badminton   
15         Atlanta                Rowing   
17         Atlanta              Swimming   
...            ...                   ...   
40604     Helsinki             Athletics   
40609       London            Volleyball   
40610       Sydney                Rowing   
40612        Sochi           Ski Jumping   
40614       Nagano             Bobsleigh   

                                          Event   Medal  
0              Speed Skating Women's 500 metres     NaN  
6      Cross Country Skiing Men's 10 kilometres     NaN  
14                      Badminton Men's Singles     NaN  
15       Rowing Men's Lightweight Double Sculls  Silver  
17          Swimming Men's 100 metres Butterfly     NaN  
...                                         ...     ...  
40604                Athletics Women's Shot Put    Gold  
40609               Volleyball Men's Volleyball     NaN  
40610    Rowing Men's Lightweight Coxless Fours     NaN  
40612  Ski Jumping Men's Large Hill, Individual     NaN  
40614                      Bobsleigh Men's Four     NaN  

[20336 rows x 12 columns]

In [23]:
one_per_athlete.ID.duplicated().sum()

0

In [25]:
# se aplica .mean() a dataframe nuevo con ID unicos
one_per_athlete.Height.mean()

176.31818479042698